### PR TITLE
wait 20 sec before delete disruptioncron + finalizer

### PR DESCRIPTION
**Pull request by clairecng**
## What does this PR do?

- [x] Adds new functionality
- [ ] Alters existing functionality
- [ ] Fixes a bug
- [ ] Improves documentation or testing

Please briefly describe your changes as well as the motivation behind them:
This PR adds a finalizer to the DisruptionCron and the finalizerDeletionDelay logic as well (wait for finalizerDeletionDelay before fully removing the finalizer).

## Code Quality Checklist

- [ ] The documentation is up to date.
- [ ] My code is sufficiently commented and passes continuous integration checks.
- [ ] I have signed my commit (see [Contributing Docs](../CONTRIBUTING.md)).

## Testing

- [ ] I leveraged continuous integration testing
    - [ ] by depending on existing `unit` tests or `end-to-end` tests.
    - [ ] by adding new `unit` tests or `end-to-end` tests.
- [ ] I manually tested the following steps:
    - `x`
    - [ ] locally.
    - [ ] as a canary deployment to a cluster.

### CODE REVIEW

File: controllers/disruption_cron_controller.go [hunk 1]
```diff
@@ -58,8 +61,37 @@ func (r *DisruptionCronReconciler) Reconcile(ctx context.Context, req ctrl.Reque
...
+
+				if err := r.Client.Update(ctx, instance); err != nil {
+					return ctrl.Result{}, fmt.Errorf("error removing disruptioncron finalizer: %w", err)
+				}
+
+				return ctrl.Result{}, nil
+			} else if instance.Status.CleanedAt == nil {
+				// set cleanedAt
+				now := metav1.Now()
+				instance.Status.CleanedAt = &now
```
[the remark this answers is not in the document]
Comment (clairecng) [reply controllers/disruption_cron_controller.go:83]: naming is hard
startedDeletionAt?

We still need it for the collection which takes 10s as well... same thing for disruptions. We can figure out another way if you think that's best though

[the remark this answers is not in the document]
Comment (ptnapoleon) [reply controllers/disruption_cron_controller.go:83]: Do we just want the deletion timestamp? If we leave the finalizer on for 10-20s after deletion, we should be able to capture that, no?

[the remark this answers is not in the document]
Comment (clairecng) [reply controllers/disruption_cron_controller.go:83]: we can do that yes. I think it's not that important actually
let me close it and discuss this on our slack

**Review by ptnapoleon — APPROVED** [review]

File: api/v1beta1/disruption_cron_types.go [hunk 2]
```diff
@@ -102,3 +104,8 @@ type DisruptionCronTrigger struct {
 	Kind      string      `json:"kind,omitempty"`
 	CreatedAt metav1.Time `json:"createdAt,omitempty"`
 }
+
+// IsReadyToRemoveFinalizer checks if adisruptioncron has been deleting for > finalizerDelay
+func (d *DisruptionCron) IsReadyToRemoveFinalizer(finalizerDelay time.Duration) bool {
```
Comment (aymericDD) [inline api/v1beta1/disruption_cron_types.go:109]: Even if it's a small method can you add unit tests? 😇

**Review by aymericDD — APPROVED** [review]

File: api/v1beta1/disruption_cron_types.go (hunk 2 above)
Comment (aymericDD) [reply api/v1beta1/disruption_cron_types.go:109]: Thanks 🙇

**Pull request merged**